src/dictionary.pl

```diff
@@ -7,31 +7,37 @@
 % A dictionary is:
 % empty
 % dict(Key, Value, Dict) where Key-Value is a pair in dict, Dict is the rest of the dictionary
-
+% a dictionary does not have duplicate keys
 
 % True if the dictionary has no values
 dictIsEmpty(empty). % only true for empty
 
 
 % True if the dictionary has the key `Key`
 % TODO: implement this (1 hour) - Charles
-hasKey(Key, dict(K,V,D)) :- notImplemented.
+hasKey(Key, dict(K,V,D)) :- hasKey(Key, D).
```

Comment:
Better to be safe and make sure `Key != K`
```suggestion
hasKey(Key, dict(K,V,D)) :- dif(Key, K), hasKey(Key, D).
```